src/dune_rules/dep_conf_eval.ml

```diff
@@ -48,11 +48,21 @@ let dep expander = function
              let pkg = Package.Name.of_string pkg in
              let context = Expander.context expander in
              match Expander.find_package expander pkg with
-             | Some pkg ->
+             | Some (Local pkg) ->
                Action_builder.alias
                  (Build_system.Alias.package_install
                     ~context:(Context.build_context context)
                     ~pkg)
+             | Some (Installed pkg) ->
+               let files =
```

Comment:
Without the check: you use dune 2.9 to work on a 2.8 project. you add a `(package x)` dependency on an external package and it works. Then you release the package and someone installs it with dune 2.8 (because your project still delcares a dependency on dune >= 2.8) and it fails because dune 2.8 doesn't support that.

With the check: you get an error at the first step and you immediately bump the `(lang dune 2.8)` line.

Reply:
Hmm, there is indeed an (interesting) open question here. I insisted on this because it feels more like a new feature than a bugfix to me.

Reply:
Now that you mention it, to be consistent we should also change the wording in the changelog file to present this as a new feature rather than a bug.